tests/test_dense.py

```diff
@@ -1,7 +1,17 @@
-from qutip_cupy import CuPyDense
+import warnings
 import numpy as np
 import pytest
 
+with warnings.catch_warnings():
+    warnings.filterwarnings(
+        action="ignore", category=UserWarning, message=r"matplotlib not found:"
+    )
+
+    from qutip.core import data
+
+from qutip_cupy import CuPyDense  # noqa:E402
+from qutip_cupy import dense  # noqa:E402
```

Comment:
```suggestion
from qutip_cupy import CuPyDense
from qutip_cupy import dense

# import qutip after qutip_cupy to supress the matplotlib import warning
from qutip.core import data
```
I think importing qutip after qutip_cupy will avoid the warning and save copying and pasting the warning filter in all the tests.